apisix/healthcheck_manager.lua

```diff
@@ -60,6 +61,9 @@ end
 
 
 local function fetch_latest_conf(resource_path)
+    -- if resource path contains json path, extract out the prefix
+    -- for eg: extracts /routes/1 from /routes/1#plugins.abc
+    resource_path = resource_path:match("^(.-)#") or resource_path
```

Comment:
use `local res = ngx.re.split(...)`

the `res[1]` is resource path, `res[2]` is sub resource path

we can add a new function to parse the path